src/plots/plots.js

```diff
@@ -563,6 +567,83 @@ function remapTransformedArrays(cd0, newTrace) {
     }
 }
 
+var formatKeys = [
+    'days', 'shortDays', 'months', 'shortMonths', 'periods',
+    'dateTime', 'date', 'time',
+    'decimal', 'thousands', 'grouping', 'currency'
+];
+
+/**
+ * getD3FormatObj: use _context to get the d3.locale argument object.
+ * decimal and thousands can be overridden later by layout.separators
+ * grouping and currency are not presently used by our automatic number
+ * formatting system but can be used by custom formats.
+ *
+ * @returns {object} d3.locale format object
+ */
+function getD3FormatObj(gd) {
+    var locale = gd._context.locale;
+    if(!locale) locale === 'en-US';
+
+    var formatDone = false;
+    var formatObj = {};
+
+    function includeFormat(newFormat) {
+        var formatFinished = true;
+        for(var i = 0; i < formatKeys.length; i++) {
+            var formatKey = formatKeys[i];
+            if(!formatObj[formatKey]) {
+                if(newFormat[formatKey]) {
+                    formatObj[formatKey] = newFormat[formatKey];
+                }
+                else formatFinished = false;
+            }
+        }
+        if(formatFinished) formatDone = true;
+    }
+
+    // same as localize, look for format parts in each format spec in the chain
+    for(var i = 0; i < 2; i++) {
+        var formats = gd._context.formats;
+        for(var j = 0; j < 2; j++) {
+            var formatj = formats[locale];
+            if(formatj) {
+                includeFormat(formatj);
+                if(formatDone) break;
+            }
+            formats = Registry.formatRegistry;
+        }
+
+        var baseLocale = locale.split('-')[0];
+        if(formatDone || baseLocale === locale) break;
+        locale = baseLocale;
+    }
+
+    // lastly pick out defaults from english (non-US, as DMY is so much more common)
```

Comment:
I strongly agree with picking `en` (not `en-us`) as our default :ok_hand:

Reply:
> picking `en` (not `en-us`) as our default

OK, so what we're doing right now is a bit schizophrenic, but it has a certain logic to it: `en-US` is the default locale, but if you specify a different locale and it has an incomplete definition (After potentially combining the region and language definitions), it will fill in with non-translated strings (for which we've use US english, `en-US`) but `en` for the missing formatting fields. But the ONLY difference between `en` and `en-US` formatting is if you make a custom `tickformat` with [`%x`](https://github.com/d3/d3-time-format) (and possibly `%c`) in it - you get DD/MM/YYYY format from `en` vs MM/DD/YYYY from `en-US`. Seemed to me that since the US is [about the only country in the world](https://en.wikipedia.org/wiki/Date_format_by_country) that uses the utterly nonsensical MM/DD/YYYY format, if you've already said you're not in the US we shouldn't fall back on that.

For strings, the only place non-translated is different is different is color/colour. I guess I don't care much about that, it currently only shows up in "Click to enter Colo(u)rscale title," and I see no legitimate reason to give an incomplete set of string translations unless you're in another english region (in which case `en` is part of the inheritance chain anyway so you'll get your silly extra "u")

Any quibbles with that logic?

Reply:
> Any quibbles with that logic?

None from me :ok_hand: